apps/projects/routes.py

```diff
@@ -89,38 +157,171 @@ def post(self):
         if codec_name not in app.config.get('CODEC_SUPPORT'):
             return bad_request("Codec: {} is not supported.".format(codec_name))
 
-        # put file into storage
+        # generate file name
         file_name = create_file_name(ext=file.filename.split('.')[1])
-        mime_type = file.mimetype
-
-        # get path group by year month
-        create_date = datetime.utcnow()
-        folder = f'{create_date.year}/{create_date.month}'
-
-        # put stream file into storage
-        if app.fs.put(file_stream, f'{folder}/{file_name}'):
+        # put file stream into storage
+        storage_id = app.fs.put(file_stream, filename=file_name, content_type=file.mimetype)
+        if storage_id:
             try:
                 # add record to database
                 doc = {
                     'filename': file_name,
-                    'folder': folder,
+                    'storage_id': storage_id,
                     'metadata': metadata,
-                    'create_time': create_date,
-                    'mime_type': mime_type,
+                    'create_time': datetime.utcnow(),
+                    'mime_type': file.mimetype,
                     'version': 1,
                     'processing': False,
                     'parent': None,
                     'thumbnails': {},
                     'client_info': user_agent,
-                    'original_filename': file.filename
+                    'original_filename': file.filename,
+                    'preview_thumbnail': None,
+                    'url': None
                 }
                 app.mongo.db.projects.insert_one(doc)
+                # create url for preview video
+                doc['url'] = app.fs.url_for_media(doc.get('_id'))
```

Comment:
That's a good idea. 
But If we use `storage_id`, I think it will make the endpoints be inconsistency https://github.com/superdesk/video-server/pull/10#discussion_r288133655
And it's not secure https://github.com/superdesk/video-server/pull/10#discussion_r287753598
@ride90 any sugess

Reply:
@idsvn I think that `url_for_media` function this is something which is outside of `fs` context. FS cares only about storing files and not generating urls.